### ISSUE TITLE
Saving of a BPMN file takes long time (dev.app)

### ISSUE THREAD

**Issue by sashayar13** [sartography/spiff-arena#1461]
I timed saving the file in the dev.app several times, and it took an average of 16 seconds.

**Comment by jasquat** [comment]
My save time seems to be down to ~2 seconds now. It probably could be better but at least it's not 16 anymore.

**Comment by madhurrya** [comment]
It is still taking time. I think this is not deployed yet?
https://drive.google.com/file/d/1l2Qkf0haFI6dRogOWB7Yjqx-4GS462eK/view?usp=sharing

**Comment by jasquat** [comment]
@madhurrya What environment were you on? On dev.app it's saving quickly for me in that model: https://dev.app.spiff.status.im/editor/process-models/misc:test:madhurya-testing:boundary-event-test/files/httpv2.bpmn

**Comment by madhurrya** [comment]
It was in dev.app. As I mentioned it seems to happen for the first time. I am not sure exactly. As you can see in the second video, for me also it saved quickly the second time.

**Comment by madhurrya** [comment]
For me its the behaviour for any random model I try. First time it takes about 20 -30 secs and next time it saves in few seconds.
https://drive.google.com/file/d/1KxDWaat_wT1YdI9mAO69zbvASWXk6ywW/view?usp=sharing

This video has the network tab opened. Didn't notice any issue there. But in the first one, the response says no data to show, but when I go to the preview, it shows data.
https://drive.google.com/file/d/13YFcZuFiiYRKcRa6IsGBy8ubjLlQHK2t/view?usp=sharing

you can see the same here also with the network tab.
https://drive.google.com/file/d/1Oj6-WWmNpzl6Hdvtf8HxUy6pLo8pSLVA/view?usp=sharing

**Comment by madhurrya** [comment]
Above are from Edge and checked from Firefox to see whether it's browser related and still noticing the same.
https://drive.google.com/file/d/1XJhtjnzixlKwQq0JaZOyKE2Eer3C4c5f/view?usp=sharing

**Comment by madhurrya** [comment]
Created a new model and saved it and it works fast.

**Comment by madhurrya** [comment]
@burnettk @jasquat I am still seeing the same behaviour for all the existing models. When I open and save it for the first time (not exactly sure what this first time is, I am opening these models after some time) it takes about 30 secs to save. But after that it saves fasts.
Can you please try opening and saving a model that you have not opened/saved in some time and see.

@sashayar13 does everything looks OK to you?

**Comment by jasquat** [comment]
@madhurrya yep I've done that and haven't seen the issue. It's 1 to 3 seconds to save.

**Comment by madhurrya** [comment]
Dinithi also didn't notice my issue. If everything looks OK to @sashayar13 I'll close this ticket.

**Comment by sashayar13** [comment]
looks a lot better - saving the changes much after - thank you!

**Comment by madhurrya** [comment]
Since I am the only one who is noticing the above behavior I'll close this ticket for now.

**Comment by madhurrya** [comment]
Released to Prod on May 20th. Prod version 18/19.
